cpu/sam0_common/include/periph_cpu_common.h

```diff
@@ -680,6 +683,35 @@ typedef struct {
  * destination, are kept in RAM and are read when the channel is enabled and
  * triggered. On the SAML21 platform, these descriptors must reside in the LP
  * SRAM.
+ *
+ * The DMA addresses supplied must point to the **end** of the array to be
+ * transferred. When address increment is enabled this means that the supplied
+ * src or dst argument must point to array + length. When increment is disabled,
+ * the source or destination address can be used directly. The calculation of
+ * the end of the array must be done by the calling function, because the
+ * beatsize and the increment can usually be hardcoded there and doesn't have to
+ * be retrieved from the DMA register configuration.
+ * See also section 20.6.2.7 of the SAM D21/DA1 Family Data Sheet.
+ *
+ * Example:
+ * ```
+ * void transfer_data(void *src, void *dst, size_t len)
+ * {
+ *      dma_t channel = dma_acquire_channel()
+ *      if (channel == 0xff) {
+ *          return -E_BUSY;
+ *      }
+ *
+ *      dma_setup(channel, DMA_TRIGGER_MY_PERIH, 0, true);
+ *      dma_prepare(channel, DMAC_BTCTRL_BEATSIZE_BYTE_Val,
+ *                  (uint8_t*)src + len, (uint8_t*)dst + len, len);
```

Comment:
You can drop the cast here too ;)

Reply:
I don't remember why I put them there in the first place 🤔

Reply:
Oh, I remember now, additions on `void*` are undefined, cppcheck will complain about this.